logstash-core/lib/logstash/api/commands/stats.rb

```diff
@@ -120,9 +151,16 @@ def report(stats)
                 :outputs => plugin_stats(stats, :outputs)
               },
               :reloads => stats[:reloads],
-              :queue => stats[:queue]
+              :queue => extended_stats["queue"]
             }.merge(stats[:dlq] ? {:dead_letter_queue => stats[:dlq]} : {})
+            if !opts[:vertices]
+              ret.delete('vertices')
+              unless extended_stats.nil?
```

Comment:
Now that we have changed L154, and given that the interface still defaults `extended_stats` to nil, do you still feel like we should remove this check?

Reply:
Please see [this comment](https://github.com/elastic/logstash/pull/10576#discussion_r292573616), which has a suggested path forward. We are adding a lot of complexity by merging our carefully-crafted `ret` into `extended_stats` and then attempting to remove the bits we don't want, and it would be simpler to explicitly compose the bits that we do want.

Doing so would also shield us from needing to mutate the objects that were passed to us.

Reply:
I have implemented your improvements to the way we compose that return. Therefore, I'm going to make this as resolved.